### PR TITLE
feat: added support for user-defined border widths in i3bar blocks

### LINKED ISSUE #3722: Support user-defined border widths per edge in i3bar protocol
## I'm submitting a…

<pre>
[ ] Bug
[x] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Desired Behavior

In i3-gaps, the i3bar protocol allows `border_[top|left|bottom|right]` keys to specify, per status block, the border of each edge in pixels. The exact behavior should be migrated into i3.

## Environment
Output of `i3 --moreversion 2>&-`:
<pre>
i3 version: 4.16.1
</pre>

### PR DESCRIPTION
**Pull request by Airblader**
This change introduces support for four new properties on the i3bar
protocol, namely "border_top", "border_right", "border_bottom"
and "border_left".

If a block is drawn with a border, these values define the width of
the corresponding edge in pixels. They all default to 1 if not
specified to preserve compatibility.

fixes #3722

### CODE REVIEW

File: i3bar/src/xcb.c [hunk 1]
```diff
@@ -287,8 +287,8 @@ static void draw_statusline(i3_output *output, uint32_t clip_left, bool use_focu
 
         color_t bg_color = bar_color;
 
-        int border_width = (block->border) ? logical_px(1) : 0;
         int full_render_width = render->width + render->x_offset + render->x_append;
+        bool is_border = !!block->border;
```
Comment (stapelberg) [inline i3bar/src/xcb.c]: Why the change in strategy here? I liked the previous solution (assigning 0 or 1 to an integer) better than relying on the fact that bool is implemented as 0 and 1.

Comment (Airblader) [reply i3bar/src/xcb.c]: I agree, I like it more as well. This is mostly just a ported patch and I wrote this years ago… :-) I've adjusted it to an explicit ternary again and also renamed it.

**Comment by Airblader** [comment]
@stapelberg done

File: i3bar/src/xcb.c [hunk 2]
```diff
@@ -287,8 +287,8 @@ static void draw_statusline(i3_output *output, uint32_t clip_left, bool use_focu
 
         color_t bg_color = bar_color;
 
-        int border_width = (block->border) ? logical_px(1) : 0;
         int full_render_width = render->width + render->x_offset + render->x_append;
+        bool has_border = block->border ? 1 : 0;
```
Comment (stapelberg) [inline i3bar/src/xcb.c]: Can you also change the type from bool to int, please?

Comment (Airblader) [reply i3bar/src/xcb.c]: @stapelberg Done
